dlp/snippets/risk.py

```diff
@@ -383,6 +383,167 @@ def callback(message: google.cloud.pubsub_v1.subscriber.message.Message) -> None
+        of a column set in a Google BigQuery table.
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        source_table_project_id: The Google Cloud project id where the BigQuery table
+            is stored.
+        source_dataset_id: The id of the dataset to inspect.
+        source_table_id: The id of the table to inspect.
+        entity_id: It enables you to accurately determine k-anonymity.
+        quasi_ids: A set of columns that form a composite key.
+        output_table_project_id: The Google Cloud project id where the output BigQuery table
+            is stored.
+        output_dataset_id: The id of the output BigQuery dataset.
+        output_table_id: The id of the output BigQuery table.
+    """
+
+    # Instantiate a client.
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Location info of the source BigQuery table.
+    source_table = {
+        "project_id": source_table_project_id,
+        "dataset_id": source_dataset_id,
+        "table_id": source_table_id,
+    }
+
+    # Specify the bigquery table to store the findings.
+    # The output_table_id in the given BigQuery dataset will be created if it doesn't
+    # already exist.
+    dest_table = {
+        "project_id": output_table_project_id,
+        "dataset_id": output_dataset_id,
+        "table_id": output_table_id,
+    }
+
+    # Convert quasi id list to Protobuf type
+    def map_fields(field: str) -> dict:
+        return {"name": field}
+
+    #  Configure column names of quasi-identifiers to analyze
+    quasi_ids = map(map_fields, quasi_ids)
+
+    # Tell the API where to send a notification when the job is complete.
+    actions = [
+        {
+            "save_findings": {
+                "output_config": {"table": dest_table}
+            }
+        }
+    ]
+
+    # Configure the privacy metric to compute for re-identification risk analysis.
+    # Specify the unique identifier in the source table for the k-anonymity analysis.
+    privacy_metric = {
+        "k_anonymity_config": {
+            "entity_id": {
+                "field": {"name": entity_id}
+            },
+            "quasi_ids": quasi_ids,
+        }
+    }
+
+    # Configure risk analysis job.
+    risk_job = {
+        "privacy_metric": privacy_metric,
+        "source_table": source_table,
+        "actions": actions,
+    }
+
+    # Convert the project id into a full resource id.
+    parent = f"projects/{project}/locations/global"
+
+    # Call API to start risk analysis job.
+    response = dlp.create_dlp_job(
+        request={
+            "parent": parent,
+            "risk_job": risk_job,
+        }
+    )
+    print("Inspection Job started : {}".format(response.name))
+
+    job_name = response.name
+
+    # Waiting for a maximum of 15 minutes for the job to be completed.
+    job = dlp.get_dlp_job(request={"name": job_name})
+    no_of_attempts = 30
+    while no_of_attempts > 0:
```

Comment:
question: is this the recommended way to wait for DLP's job results? It seems quite cumbersome to poll the result every 30 seconds. Does the library support long-running operations in similar way as compute does ([see here](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/compute/client_library/snippets/operations/wait_for_extended_operation.py) for an example of handling an LRO).

Reply:
Yes, I also feel this looks quite cumbersome. Here, we are calling get_dlp_job, which will give us the response at that instant, does it account to a long-running jobs? create_dlp_job or get_dlp_job, both of them doesn't seem to be long running operation. I couldn't find any efficient way to poll the request till we get the status - "Done". Though I am still looking at it and will update the code accordingly if I find any good way to handle this. Thank you!

Reply:
@realjordanna Do you have any recommendations? Are there any convenience methods in the SDK that help. This has come up in the past as well.